### PR TITLE
Add Protobuf sampler and hooks for SCollection

### PR DESCRIPTION
**Pull request by idreeskhan**
Allows user to import and use Sampler as a lib more easily and adds functionality for Protobuf

### CODE REVIEW

File: ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala [hunk 1]
```diff
@@ -319,27 +329,103 @@ private[samplers] trait BigSampler {
...
     }
   }
   //scalastyle:on method.length cyclomatic.complexity
+
+  def run(argv: Array[String]): Unit = {
+    this.singleInput(argv)
+  }
+}
+
+private[samplers] object BigSamplerProtobuf {
```
[the remark this answers is not in the document]
Comment (idreeskhan) [reply ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala]: 👍

File: ratatool-sampling/README.md [hunk 2]
```diff
@@ -43,7 +44,37 @@ The following options are recommended, but may not be necessary.
...
+* @param fields Fields to construct hash over for determinism
+* @param seed Seed used to salt the deterministic hash
+* @param distribution Desired output sample distribution
+* @param distributionFields Fields to construct distribution over (strata = set of unique fields)
+* @param precision Approximate or Exact precision
+* @param maxKeySize Maximum allowed size per key (can be tweaked for very large data sets)
+* @tparam T Record Type
+* @return SCollection containing Sample population
+*/
+def sampleProto[T <: AbstractMessage : ClassTag](coll: SCollection[T],
```
[the remark this answers is not in the document]
Comment (jbigred1) [reply ratatool-sampling/README.md]: well I mean in the documentation just to show that you can sample either directly

File: ratatool-sampling/README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ For full details see [BigSample.scala](https://github.com/spotify/ratatool/blob/
 
 ## Usage
 
+### Command Line
 ```
 BigSampler - a tool for big data sampling
 Usage: ratatool bigSampler [dataflow_options] [options]
```
Comment (jbigred1) [inline ratatool-sampling/README.md]: do we need to add something here about Protobuf as well

Comment (idreeskhan) [reply ratatool-sampling/README.md]: No, similar to BigDiffy, proto from CLI is not so simple.

**Pull request merged**